src/mlpack/tests/pca_test.cpp

```diff
@@ -317,25 +314,22 @@ BOOST_AUTO_TEST_CASE(PCAScalingTest)
   // The first two components of the eigenvector with largest eigenvalue should
   // be somewhere near sqrt(2) / 2.  The third component should be close to
   // zero.  There is noise, of course...
-  BOOST_REQUIRE_CLOSE(std::abs(eigvec(0, 0)), sqrt(2) / 2, 0.35);
-  BOOST_REQUIRE_CLOSE(std::abs(eigvec(1, 0)), sqrt(2) / 2, 0.35);
-  BOOST_REQUIRE_SMALL(eigvec(2, 0), 0.1); // Large tolerance for noise.
+  REQUIRE(std::abs(eigvec(0, 0)) == Approx(sqrt(2) / 2).epsilon(0.0035));
+  REQUIRE(std::abs(eigvec(1, 0)) == Approx(sqrt(2) / 2).epsilon(0.0035));
+  REQUIRE(eigvec(2, 0) == Approx(0.0).margin(0.1)); // Large tolerance for noise.
 
   // The second component should be focused almost entirely in the third
   // dimension.
-  BOOST_REQUIRE_SMALL(eigvec(0, 1), 0.1);
-  BOOST_REQUIRE_SMALL(eigvec(1, 1), 0.1);
-  BOOST_REQUIRE_CLOSE(std::abs(eigvec(2, 1)), 1.0, 0.35);
+  REQUIRE(eigvec(0, 1) == Approx(0.0).margin(0.1));
+  REQUIRE(eigvec(1, 1) == Approx(0.0).margin(0.1));
+  REQUIRE(std::abs(eigvec(2, 1)) == Approx(1.0).epsilon(0.0035));
 
   // The third component should have the same absolute value characteristics as
   // the first (plus 20% tolerance).
-  BOOST_REQUIRE_CLOSE(std::abs(eigvec(0, 0)), sqrt(2) / 2, 0.35);
```

Comment:
Like in the comment below, here the tolerance is equal to 0.35% (according to instructions in #2523). What should we do here? Change comment or change code? Tests pass so I guess comment should be changed

Reply:
@mikkac I think the comment got out of date---it should say 0.35% like you pointed out.  My guess is that it started at 0.2, and whoever wrote the comment made an error thinking that represented 20%, then later the tolerance was increased to reduce random failures but the comment was forgotten. :smile: Thanks for noticing and pointing it out! :+1: